apps/contrib/templatetags/contrib_tags.py

```diff
@@ -29,3 +30,13 @@ def concat_strings(*args):
     for string in args:
         concat_str += str(string)
     return concat_str
+
+
+@register.assignment_tag
+def get_absolute_uri(request, obj):
```

Comment:
Passing the `request` parameter might not be necessary here. You can instead use `@register.simple_tag(takes_context=True)`, which will give you the template context. The request is than available from `context['request']`. This works because we have the [request context processor](https://docs.djangoproject.com/en/1.11/ref/templates/api/#using-requestcontext) active.